Pipeline Syntax page - Document that parameters become environment variables (#5919)
jenkins-infra/jenkins.io: content/doc/book/pipeline/syntax.adoc

parameters are made available to Pipeline steps via the `params` object,
see the <<parameters-example>> for its specific usage.

Each parameter has a _Name_ and some sort of _Value_, depending on the parameter type.
These name-value pairs will be exported as environment variables when the build starts, allowing subsequent parts of the build configuration (such as build steps) to access those values, e.g. by using the `+{PARAMETER_NAME}+` syntax (or `%PARAMETER_NAME%` on Windows).

> 
> [NOTE]
> ====
> This also implies that each parameter defined here should have a unique Name. For more information visit 
> link:https://github.com/jenkinsci/jenkins/blob/8a29e8858b49da45c0071f690d4ceebe8db2f18f/core/src/main/resources/hudson/model/ParametersDefinitionProperty/help.html[ParametersDefinitionProperty]
> ====

Reviewer: I think that the detail offered in this note is too broad for the section of documentation where it is placed and the linked page is not a usable location for most readers.  Most readers do not expect to read HTML markup in order to understand a concept.

I recommend it be removed from the pull request.
```suggestion

```

Author: Thanks @MarkEWaite for the review. The HTML markup contains information i think that is relevant to this PR maybe i can copy paste it somewhere where it can be seen by readers.
and the note can be shorten without referencing the link.
What are your thoughts on that?

Author: Thanks @MarkEWaite for the review. I can shorten the note and remove the link.

parameters are made available to Pipeline steps via the `params` object,
see the <<parameters-example>> for its specific usage.

> Each parameter has a _Name_ and some sort of _Value_, depending on the parameter type.
> These name-value pairs will be exported as environment variables when the build starts, allowing subsequent parts of the build configuration (such as build steps) to access those values, e.g. by using the `+{PARAMETER_NAME}+` syntax (or `%PARAMETER_NAME%` on Windows).
> 

Reviewer 2: ```suggestion
Each parameter has a _Name_ and  _Value_, which depends on the parameter type.
These name-value pairs are exported as environment variables when the build starts, allowing subsequent parts of the build configuration, such as build steps, to access those values.
For example, this can be performed by using the `+{PARAMETER_NAME}+` syntax (or `%PARAMETER_NAME%` on Windows).

```
small syntax/wording suggestion, let me know if you have any questions!

Author: Thanks @kmartens27 for the suggestion.
The wording seems much clear and easy to understand now.

parameters are made available to Pipeline steps via the `params` object,
see the <<parameters-example>> for its specific usage.

> Each parameter has a _Name_ and  _Value_, which depends on the parameter type.
> These name-value pairs are exported as environment variables when the build starts, allowing subsequent parts of the build configuration, such as build steps, to access those values.
> For example, this can be performed by using the `+{PARAMETER_NAME}+` syntax (or `%PARAMETER_NAME%` on Windows).

Reviewer 2: ```suggestion
Each parameter has a _Name_ and _Value_, depending on the parameter type.
This information is exported as environment variables when the build starts, allowing subsequent parts of the build configuration to access those values.
For example, this can be performed by using the `+{PARAMETER_NAME}+` syntax (or `%PARAMETER_NAME%` on Windows).
```
I found an extra space, and wanted to suggest something other than name-value pairs for the parameters. If that does not convey the same message, I would be happy to discuss other options for that wording.

Author: Thanks @kmartens27 for the suggestion.I have used the words _Name_ and _Value_ because they were stated in [here](https://github.com/jenkinsci/jenkins/blob/8a29e8858b49da45c0071f690d4ceebe8db2f18f/core/src/main/resources/hudson/model/ParametersDefinitionProperty/help.html#L11-L15) however we can always discuss for other options to improve the readability. Do you have any suggestions?

Reviewer 2: Right, sorry about that!
Based on that, the name-value pairs should be fine. Disregard that part of the suggestion! Thanks very much @Vandit1604
